portal/src/graphql/portal/SubscriptionScreen.tsx

```diff
@@ -205,192 +168,305 @@ function SubscriptionPlanCardRenderer(props: SubscriptionPlanCardRenderProps) {
-  const { name } = subscriptionPlan;
+  const onPlanAction = useMemo(() => {
+    const plans: Plan[] = ["enterprise"];
+    if (subscriptionPlans.findIndex((p) => p.name === "developers") !== -1) {
+      plans.push("developers");
+    }
+    if (subscriptionPlans.findIndex((p) => p.name === "business") !== -1) {
+      plans.push("business");
+    }
+    plans.push("enterprise");
+
+    return plans.reduce<Partial<Record<Plan, (action: CTAVariant) => void>>>(
+      (callbacks, plan) => {
+        const fn = (action: CTAVariant) => {
+          switch (action) {
+            case "subscribe":
+              onClickSubscribe(plan);
+              break;
+            case "upgrade":
+              onClickUpgrade(plan);
+              break;
+            case "downgrade":
+              onClickDowngrade(plan);
+              break;
+            case "reactivate":
+              onClickReactivate();
+              break;
+            case "contact-us":
+              onClickContactUs();
+              break;
+            case "current":
+            case "non-applicable":
+            default:
+              console.error(
+                `action button clicked but action:${action} should not be clickable. plan: ${plan}`
+              );
+              break;
+          }
+        };
+        callbacks[plan] = fn;
+        return callbacks;
+      },
+      {}
+    );
+  }, [
+    onClickContactUs,
+    onClickDowngrade,
+    onClickReactivate,
+    onClickSubscribe,
+    onClickUpgrade,
+    subscriptionPlans,
+  ]);
 
-  const basePrice = subscriptionPlan.prices.find(
-    (price) => price.type === SubscriptionItemPriceType.Fixed
-  );
-  const northAmericaSMSPrice = subscriptionPlan.prices.find(
-    (price) =>
-      price.type === SubscriptionItemPriceType.Usage &&
-      price.usageType === SubscriptionItemPriceUsageType.Sms &&
-      price.smsRegion === SubscriptionItemPriceSmsRegion.NorthAmerica
-  );
-  const otherRegionsSMSPrice = subscriptionPlan.prices.find(
-    (price) =>
-      price.type === SubscriptionItemPriceType.Usage &&
-      price.usageType === SubscriptionItemPriceUsageType.Sms &&
-      price.smsRegion === SubscriptionItemPriceSmsRegion.OtherRegions
-  );
-  const northAmericaWhatsappPrice = subscriptionPlan.prices.find(
-    (price) =>
-      price.type === SubscriptionItemPriceType.Usage &&
-      price.usageType === SubscriptionItemPriceUsageType.Whatsapp &&
-      price.whatsappRegion === SubscriptionItemPriceWhatsappRegion.NorthAmerica
-  );
-  const otherRegionsWhatsappPrice = subscriptionPlan.prices.find(
-    (price) =>
-      price.type === SubscriptionItemPriceType.Usage &&
-      price.usageType === SubscriptionItemPriceUsageType.Whatsapp &&
-      price.whatsappRegion === SubscriptionItemPriceWhatsappRegion.OtherRegions
-  );
-  const mauPrice = subscriptionPlan.prices.find(
-    (price) =>
-      price.type === SubscriptionItemPriceType.Usage &&
-      price.usageType === SubscriptionItemPriceUsageType.Mau
```

Comment:
I've removed most of these codes because I think they are over complicated, and most of the time we still need to update the code when changing the billing details. So I hard coded the prices in the ui.